### PR TITLE
RocketCDN code improvements

### PR DESCRIPTION
**Pull request by Tabrisrp**
- Add method to preserve the authorization token
- Add label for cname field
- Remove prefix on rocketCDN JS functions
- Various code improvements

### CODE REVIEW

File: inc/classes/subscriber/CDN/RocketCDN/DataManagerSubscriber.php [hunk 1]
```diff
@@ -79,6 +79,10 @@ public function update_user_token() {
 			return;
 		}
 
+		if ( ! is_string( $_POST['value'] ) ) {
+			wp_send_json_error( 'invalid_token' );
```
Comment (Screenfeed) [inline inc/classes/subscriber/CDN/RocketCDN/DataManagerSubscriber.php]: No `return;` this time?

File: inc/classes/subscriber/CDN/RocketCDN/DataManagerSubscriber.php [hunk 2]
```diff
@@ -50,7 +50,7 @@ public static function get_subscribed_events() {
 			'wp_ajax_rocketcdn_process_set'    => 'set_process_status',
 			'wp_ajax_rocketcdn_process_status' => 'get_process_status',
 			self::CRON_EVENT                   => 'maybe_disable_cdn',
-			'update_option_' . WP_ROCKET_SLUG  => [ 'maybe_update_api_status', 12, 2 ],
+			'update_option_' . rocket_get_constant( 'WP_ROCKET_SLUG' ) => [ 'maybe_update_api_status', 12, 2 ],
```
Comment (Screenfeed) [inline inc/classes/subscriber/CDN/RocketCDN/DataManagerSubscriber.php]: Now the signs are not aligned anymore :s

File: inc/classes/CDN/RocketCDN/APIClient.php [hunk 3]
```diff
@@ -278,4 +278,41 @@ public function update_website_status( $is_active ) {
...
+	 * @return array
+	 */
+	public function preserve_authorization_token( $args, $url ) {
+		if ( strpos( $url, self::ROCKETCDN_API ) === false ) {
+			return $args;
+		}
+
+		if ( empty( $args['headers']['Authorization'] ) ) {
+			return $args;
+		}
```
Comment (Screenfeed) [inline inc/classes/CDN/RocketCDN/APIClient.php]: So if a plugin completely removes the header...
Shouldn't we add it back anyway?

File: inc/classes/CDN/RocketCDN/APIClient.php [hunk 4]
```diff
@@ -78,7 +78,7 @@ private function get_remote_subscription_data() {
 		}
 
 		$data = json_decode( $data, true );
-		$data = array_intersect_key( $data, $default );
+		$data = array_intersect_key( (array) $data, $default );
```
Comment (Screenfeed) [inline inc/classes/CDN/RocketCDN/APIClient.php]: To make sure to not trigger a notice when doing `if ( 'running' === $subscription_data['subscription_status'] ) {`:
```suggestion
		$data = array_intersect_key( (array) $data, $default );
		$data = array_merge( $default, $data );
```

File: inc/classes/subscriber/CDN/RocketCDN/DataManagerSubscriber.php (hunk 2 above)
Comment (Tabrisrp) [reply inc/classes/subscriber/CDN/RocketCDN/DataManagerSubscriber.php]: The sniffers are not triggered on that one, I think the structure of the key throw them off

**Pull request merged**